twoliter/src/cargo_make.rs

```diff
@@ -0,0 +1,213 @@
+use std::path::PathBuf;
+
+use anyhow::{bail, Result};
+use log::trace;
+use tokio::process::Command;
+
+use crate::common::exec;
+use crate::docker::ImageArchUri;
+use crate::project::Project;
+
+fn require_sdk(project: &Project, arch: &str) -> Result<(ImageArchUri, ImageArchUri)> {
+    match (project.sdk(arch), project.toolchain(arch)) {
+        (Some(s), Some(t)) => Ok((s, t)),
+        _ => bail!(
+            "When using twoliter make, it is required that the SDK and toolchain be specified in \
+            Twoliter.toml"
+        ),
+    }
+}
+
+#[derive(Debug, Clone, Default)]
+pub struct CargoMake {
+    makefile_path: Option<PathBuf>,
+    project_dir: Option<PathBuf>,
+    args: Vec<String>,
+}
+
+impl CargoMake {
+    pub fn new<S>(project: &Project, arch: S) -> Result<Self>
+    where
+        S: Into<String>,
```

Comment:
I keep trying Into<String> and regretting it because more things are` AsRef<str>` than `Into<String>` and for some reason there is no blanket impl that goes from `AsRef<str>` to `Into<String>`.

```suggestion
        S: AsRef<str>,
```

Reply:
Since we are taking ownership of each `String` it makes since to use `Into<String>`. We can change this if it's critical, but I believe `Into<String>` is the correct type here.